src/anyio/_core/_tasks.py

```diff
@@ -156,3 +180,248 @@ def create_task_group() -> TaskGroup:
 
     """
     return get_async_backend().create_task_group()
+
+
+class TaskHandle(Generic[T]):
+    __slots__ = (
+        "cancel_scope",
+        "_event",
+        "_return_value",
+        "_exception",
+        "_start_value",
+    )
+
+    _return_value: T
+    _start_value: Any
+
+    def __init__(self) -> None:
+        from ._synchronization import Event
+
+        self.cancel_scope: CancelScope = CancelScope()
+        self._event = Event()
+        self._exception: BaseException | None = None
+
+    def set_return_value(self, value: T) -> None:
+        self._return_value = value
+        self._event.set()
+
+    def set_exception(self, exception: BaseException) -> None:
+        self._exception = exception
+        self._event.set()
+
+    def cancel(self) -> None:
+        self.cancel_scope.cancel()
+
+    @property
+    def start_value(self) -> Any:
```

Comment:
It seems that it's not possible to await the start value, should this be an async function instead?

Reply:
My mistake, see https://github.com/agronholm/anyio/pull/890/files#r2006131909.